kernel/sched.c

```diff
@@ -975,6 +977,8 @@ s32_t z_impl_k_sleep(int ms)
 {
 	s32_t ticks;
 
+	__ASSERT(ms >= 0, "Only positive values are accepted.");
```

Comment:
Can't do this.  Existing code passes `K_FOREVER` which will fail.  That needs to get cleaned up.

Reply:
in the tree?

Reply:
Yes.  Ten examples in-tree, probably more outside.  `k_sleep(K_FOREVER)` is an obvious thing to do, even though it doesn't work as people expect unless it's within a `while (true)` loop.

Reply:
moved it to #20448.